query-algebrizer/src/clauses/resolve.rs

```diff
@@ -65,6 +66,36 @@ impl ConjoiningClauses {
         }
     }
 
+    /// Just like `resolve_numeric_argument`, but for `ValueType::Instant`.
+    pub fn resolve_instant_argument(&mut self, function: &PlainSymbol, position: usize, arg: FnArg) -> Result<QueryValue> {
+        use self::FnArg::*;
+        match arg {
+            FnArg::Variable(var) => {
+                self.constrain_var_to_type(var.clone(), ValueType::Instant);
+                self.column_bindings
+                    .get(&var)
+                    .and_then(|cols| cols.first().map(|col| QueryValue::Column(col.clone())))
+                    .ok_or_else(|| Error::from_kind(ErrorKind::UnboundVariable(var.name())))
+            },
+            Constant(NonIntegerConstant::Instant(v)) => {
+                Ok(QueryValue::TypedValue(TypedValue::Instant(v)))
+            },
+
+            // TODO: should we allow integers if they seem to be timestamps? It's ambiguous…
```

Comment:
Generally, I'd prefer Mentat to be strict (with errors rather than empty result sets) in ambiguous cases.